sqladmin/application.py

```diff
@@ -182,41 +184,70 @@ def _handle_expose_decorated_func(
 
             view.identity = getattr(func, "_identity")
 
-    def add_model_view(self, view: Type[ModelView]) -> None:
-        """Add ModelView to the Admin.
+    def get_sessionmaker(
+        self,
+    ) -> tuple[sessionmaker[Session] | sessionmaker[AsyncSession], bool]:
+        """
+        Get a sessionmaker for a SQLAlchemy session and whether it is async or not.
 
         ???+ usage
             ```python
-            from sqladmin import Admin, ModelView
-
-            class UserAdmin(ModelView, model=User):
-                pass
+            from sqladmin import Admin
+
+            class MyAdmin(Admin):
+                def __init__(self, *args, **kwargs):
+                    super().__init__(*args, **kwargs)
+                    self.admin.add_route(
+                        "/login_oauth",
+                        self.login_oauth,
+                        methods=["GET", "POST"],
+                    )
 
-            admin.add_model_view(UserAdmin)
+                async def login_oauth(self, request: Request) -> Response:
+                    ...
+                    sessionmaker, _ = self.get_sessionmaker()
```

Comment:
Why can't we access it like this?

```py
async def login_oauth(self, request: Request) -> Response:
    with self.sessionmaker() as session:
        ...
```

Reply:
I think the reason I'm confused is that since we are using session maker instead of using a session,
our new route doesn't need a new session maker, a single session maker should be enough for all routes, right?
They can just get new sessions from the sessionmaker.

Reply:
Are suggesting that we set a `sessionmaker` attribute on the `Admin` class? I am subclassing `Admin` not `View`. A `View` has a `sessionmaker` attribute but not the `Admin` class.

Reply:
You're right I missed that part.

I think need to move this: https://github.com/aminalaee/sqladmin/blob/978cdd6587c7fc8345076393ae6774cddbdcd48c/sqladmin/application.py#L203

To the Admin here: https://github.com/aminalaee/sqladmin/blob/978cdd6587c7fc8345076393ae6774cddbdcd48c/sqladmin/application.py#L66

And in the view only do:

```py
view.sessionmaker = self.sessionmaker
view.async_engine = False
```

Reply:
Good idea, I have updated it to do that